docs/en/developer/udf_develop_guide.md

```diff
@@ -151,7 +151,7 @@ For more UDF implementation, see [here](../../../src/examples/test_udf.cc).
 - Run the compiling command. `-I` specifies the path of `include` directory. `-o` specifies the name of the dynamic library.
 
 ```shell
-g++ -shared -o libtest_udf.so examples/test_udf.cc -I /work/OpenMLDB/include -std=c++11 -fPIC
+g++ -shared -o libtest_udf.so examples/test_udf.cc -I /work/OpenMLDB/include -std=c++17 -fPIC
```

Comment:
reason ? our header uses c++17 features ?

Reply:
![image](https://github.com/4paradigm/OpenMLDB/assets/24697960/b544ef12-2015-4b6b-ad3c-ff242fc01475)

Reply:
ok. Maybe I can fixes the header though... still allow compile from c++11